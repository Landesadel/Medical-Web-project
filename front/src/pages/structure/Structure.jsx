import React from 'react';
import styles from "./Structure.module.scss";

//Председатель общества
const Chief = {
    first_name: "",
    surname: "",
    last_name: ""
};
//Заместитель председателя
const subChief = {
    first_name: "",
    surname: "",
    last_name: ""
};
//Секретарь правления
const administrationChief = {
    first_name: "",
    surname: "",
    last_name: ""
};
//Члены правления
const membersOfAdministration = [
    {
        first_name: "",
        surname: "",
        last_name: ""
    },
    {
        first_name: "",
        surname: "",
        last_name: ""
    },
];
//Члены ревизионной комиссии
const membersOfCommission = [
    {
        first_name: "",
        surname: "",
        last_name: ""
    },
    {
        first_name: "",
        surname: "",
        last_name: ""
    },
    {
        first_name: "",
        surname: "",
        last_name: ""
    },
    {
        first_name: "",
        surname: "",
        last_name: ""
    }
];
//Структура общества
const structure = [
    {
        title: "Председатель",
        personal_data:Chief,
        content:null
    },
    {
        title: "Заместитель председателя",
        personal_data:subChief,
        content:null
    },
    {
        title: "Правление",
        personal_data:null,
        content: [
            {
                title: "Секретарь правления",
                members:[administrationChief],
            },
            {
                title: "Члены правления",
                members: membersOfAdministration
            },
        ]
    },
    {
        title: "Ревизионная комиссия",
        personal_data:null,
        content: [
            {
                title: "Члены ревизионной комиссии",
                members: membersOfCommission
            }
        ]

    }
];


const Structure = () => {
    return (
        <div className="container">

            <h1 className={styles.heading}>{'Структура общества'}</h1>
            <div className={styles.tree}>
                <ul>
                    {structure?.map((item, idx) => (
                        <li key={idx}>
                            <div className={styles.structureBtn}>
                                <h4>{item?.title}</h4>
                                {item?.personal_data &&<p className={styles.name}>{`${item?.personal_data.last_name} ${item?.personal_data.first_name} ${item?.personal_data.surname}`}</p>}
                                {(idx !== 0 && idx !== 1) && <div className={styles.branchWrapper}>
                                    {item?.content.map((branch, idx) => (
                                        <div key={idx} className={`${styles.branch} ${styles.structureBtn}`}>
                                            <h4>{branch?.title}</h4>
                                            <div>{
                                                branch?.members.map((item, idx) => (
                                                    <p key={idx}>{`${item?.last_name} ${item?.first_name} ${item?.surname}`}</p>
                                                ))
                                            }</div>
                                        </div>

                                    ))}
                                </div>
                                }
                            </div>
                        </li>
                    ))}

                </ul>
            </div>
        </div>

    );
};

export default Structure;