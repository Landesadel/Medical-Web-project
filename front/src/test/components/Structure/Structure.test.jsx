import React from 'react';
import { render, screen} from '@testing-library/react';
import '@testing-library/jest-dom';
import Structure from "../../../pages/structure/Structure";


describe('Structure', () => {
    it('render structure', () => {
        render(
            <Structure/>
        );
        expect(screen.getByText('Ревизионная комиссия')).toBeInTheDocument();
        expect(screen.getByRole('list')).toBeInTheDocument();
    });
    it('snapshot structure', () => {
        const snapshot=render(
            <Structure/>
        );
        expect(snapshot).toMatchSnapshot();

    });
});
