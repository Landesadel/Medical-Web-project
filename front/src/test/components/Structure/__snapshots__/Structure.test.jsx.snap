// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Structure snapshot structure 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <h1
          class="heading"
        >
          Структура общества
        </h1>
        <div
          class="tree"
        >
          <ul>
            <li>
              <div
                class="structureBtn"
              >
                <h4>
                  Председатель
                </h4>
                <p
                  class="name"
                >
                    
                </p>
              </div>
            </li>
            <li>
              <div
                class="structureBtn"
              >
                <h4>
                  Заместитель председателя
                </h4>
                <p
                  class="name"
                >
                    
                </p>
              </div>
            </li>
            <li>
              <div
                class="structureBtn"
              >
                <h4>
                  Правление
                </h4>
                <div
                  class="branchWrapper"
                >
                  <div
                    class="branch structureBtn"
                  >
                    <h4>
                      Секретарь правления
                    </h4>
                    <div>
                      <p>
                          
                      </p>
                    </div>
                  </div>
                  <div
                    class="branch structureBtn"
                  >
                    <h4>
                      Члены правления
                    </h4>
                    <div>
                      <p>
                          
                      </p>
                      <p>
                          
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </li>
            <li>
              <div
                class="structureBtn"
              >
                <h4>
                  Ревизионная комиссия
                </h4>
                <div
                  class="branchWrapper"
                >
                  <div
                    class="branch structureBtn"
                  >
                    <h4>
                      Члены ревизионной комиссии
                    </h4>
                    <div>
                      <p>
                          
                      </p>
                      <p>
                          
                      </p>
                      <p>
                          
                      </p>
                      <p>
                          
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <h1
        class="heading"
      >
        Структура общества
      </h1>
      <div
        class="tree"
      >
        <ul>
          <li>
            <div
              class="structureBtn"
            >
              <h4>
                Председатель
              </h4>
              <p
                class="name"
              >
                  
              </p>
            </div>
          </li>
          <li>
            <div
              class="structureBtn"
            >
              <h4>
                Заместитель председателя
              </h4>
              <p
                class="name"
              >
                  
              </p>
            </div>
          </li>
          <li>
            <div
              class="structureBtn"
            >
              <h4>
                Правление
              </h4>
              <div
                class="branchWrapper"
              >
                <div
                  class="branch structureBtn"
                >
                  <h4>
                    Секретарь правления
                  </h4>
                  <div>
                    <p>
                        
                    </p>
                  </div>
                </div>
                <div
                  class="branch structureBtn"
                >
                  <h4>
                    Члены правления
                  </h4>
                  <div>
                    <p>
                        
                    </p>
                    <p>
                        
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </li>
          <li>
            <div
              class="structureBtn"
            >
              <h4>
                Ревизионная комиссия
              </h4>
              <div
                class="branchWrapper"
              >
                <div
                  class="branch structureBtn"
                >
                  <h4>
                    Члены ревизионной комиссии
                  </h4>
                  <div>
                    <p>
                        
                    </p>
                    <p>
                        
                    </p>
                    <p>
                        
                    </p>
                    <p>
                        
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
